framework/src/engine/consensus/consensus.ts

```diff
@@ -61,13 +61,15 @@ import { ValidatorInfo } from './certificate_generation/types';
 import { BFTModule } from '../bft';
 import { ABI, TransactionExecutionResult, TransactionVerifyResult } from '../../abi';
 import { isEmptyConsensusUpdate } from './utils';
+import { LegacyChainHandler } from '../legacy/legacy_chain_handler';
```

Comment:
i think now we don't need LegacyChainHandler for anything in consensus

Reply:
I think it was mentioned in the isse `Pass legacyChainHandler to consensus so that when consensus calls network.applyNodeInfo it can pass current legacy blocks`

Reply:
I just updated the issue by removing this requirement - Since we can apply partial nodeInfo we don't have this requirement

Reply:
ok, I already updated the PR to remove it.